### ISSUE TITLE
Support Responder initiated message in a secure session

### ISSUE THREAD

**Issue by jyao1** [DMTF/libspdm#498]
From https://github.com/DMTF/libspdm/discussions/224

Currently, the libspdm follows {requester, responder} message pair. A requester always initiates the message, then the responder gives response.

In a secure session, the responder can act as an initiator to send out message. It highly depends upon the transport layer interface (out of scope of SPDM).
How do we want to support this case?

For example, the PLDM message for firmware update (DSP0267).
@jcahill0, do we only cover DSP0267 - 1.1.0

NOTE: Not all protocol support Responder initiated message.

**Comment by jyao1** [comment]
1/10/2022 - not sure what is real use case. PLDM ?

**Comment by jcahill0** [comment]
Yes, PLDM firmware update is a major reason to support this.   In many configs, the BMC will fulfill the role as the SPDM Requestoer with one application/service/library, and as the PLDM management controller (MC).  A managed device on the transport topology will fulfill the role as SPDM Responder and PLDM management device.

So the BMC and device will perform attestation, create a session.  The a user can initiate the BMC to update the device with PDLM FW Update.  The BMC builds a PLDM message to start this, it is secured through the session.


After decoding the message, the device sends PLDM requests to pull information about the PLDM image components available, which are secured through the session.  Then when matches are found, the device sends PLDM requests to pull the actual image components for flashing/updating.

Another standardized use case will be PLDM's Control Monitoring protocol, which allows the device to asynchronously send PDLM events to the BMC, which fundamentally are PDLM requests.
